content/en/blogs/Will Cloud Native WebAssembly Replace Docker_.md

```diff
@@ -0,0 +1,90 @@
+
+This article is based on the share by Second State CEO Michael Yuan in 2020 KubeSphere Meetup.
+
+## ***Background***
+
+![Image text](https://raw.githubusercontent.com/VeraXIE1997/test/main/Will%20Cloud%20Native%20Replace%20Docker/image/image-20210721141953114%E7%9A%84%E5%89%AF%E6%9C%AC.png)
+
+Solomon Hykes, the founder of Docker in March 2019, claimed on twitter that “If WASM+WASI existed in 2008, we would not have needed to create Docker”. From where he stood, WebAssembly reveals the future of computing. This twitter led to a great stir in communities, as many people got confused about why WebAssembly, a replacement of JavaScript in browser for game playing, turned out to be an alternative to Docker in server.
+
+## *Where WebAssembly Lies in Server*
+
+![Image text](https://raw.githubusercontent.com/VeraXIE1997/test/main/Will%20Cloud%20Native%20Replace%20Docker/image-20210721142129175%E7%9A%84%E5%89%AF%E6%9C%AC.png)
+
+We tend to divide container, virtual machine or operation environment into three different levels in terms of server
+
+1. The bottom lies Hypervisor VM of hardware, or AWS Firecrackers, which is known as micro VMs with direct connection with hardware.
+
+
+2. **The middle lies Application containers VM, where application containers such as Docker can be created. Notice: application container is still within the level of operation system, so the whole operation system should be fully included.**
+
+3. The upper lies high-level language VMs that starts from JVM. WebAssembly is abstracted from the level of operating system. This is where WebAssembly lies in server.
+
+If WebAssembly manages to create a language VM similar to JVM, the dream proposed by Java over 20 years ago can be realized: to provide developers with a secure and highly abstracted operation environment across different operating systems, hardware and software platforms.
+
+## *Comparison between WebAssembly and Docker*
+
+![Image text](https://raw.githubusercontent.com/VeraXIE1997/test/main/Will%20Cloud%20Native%20Replace%20Docker/image-20210721142326218%E7%9A%84%E5%89%AF%E6%9C%AC.png)
+
+What is the relationship between WebAssembly and Docker? Why is it believed that WebAssembly will replace Docker? Compared with Docker, WebAssembly performs better in the following aspects.
+
+In terms of cold start, WebAssembly is 100 times faster than Docker.
+
+Slow cold start is a long haunting problem confronted by serverless and container service. As AWS provides reserve instances, keep it hot will offset advantages brought by serverless. Reserve will lead to charge by day, which goes against charge by millisecond we expect. One prominent edge of WebAssembly is no need to start the whole operating system, making it 100 times faster than Docker’scold start.
```

Comment:
@FeynmanZhou  Please look over this paragraph. It confused me, both in English and Chinese.

Reply:
@JiannaLiu In my opinion, it is necessary to understand the definition of `cold start`, then you can easily understand what is this paragraph meaning. Take a look at this [explanation](https://www.jianshu.com/p/a10b1e9f8f57), it might be helpful.

Reply:
thanks for the material, I will take a look. My point is that please check if the translation expresses the meaning correctly as it sounds a little unnatural to me.  @VeraXIE1997   keeping it hot...